.github/workflows/build_cmake.yml

```diff
@@ -0,0 +1,333 @@
+name: CMake Build Matrix
+
+on: [push, pull_request]
+
+env:
+  CMAKE_VERSION: 3.16.2
+  NINJA_VERSION: 1.9.0
+  BUILD_TYPE: Release
+  CCACHE_VERSION: 3.7.7
+  HDF5_VERSION: "1.10.4"
+  HDF5_VSVERSION: "14"
+  HDF5_DIR: "C:\\hdf5\\1.8.17\\VS14"
+jobs:
+  build:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {
+            name: "Windows Latest MSVC", artifact: "Windows-MSVC.tar.xz",
+            os: windows-latest,
+            cc: "cl", cxx: "cl",
+            environment_script: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
+          }
+        # - {
+        #     name: "Windows Latest MinGW", artifact: "Windows-MinGW.tar.xz",
+        #     os: windows-latest,
+        #     cc: "gcc", cxx: "g++"
+        #   }
+        # - {
+        #     name: "Ubuntu Latest GCC", artifact: "Linux.tar.xz",
+        #     os: ubuntu-latest,
+        #     cc: "gcc", cxx: "g++"
+        #   }
+        # - {
+        #     name: "macOS Latest Clang", artifact: "macOS.tar.xz",
+        #     os: macos-latest,
+        #     cc: "clang", cxx: "clang++"
+        #   }
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/checkout@v2
+    - name: Checkout submodules
+      shell: bash
+      run: |
+        # If your submodules are configured to use SSH instead of HTTPS please uncomment the following line
+        # git config --global url."https://github.com/".insteadOf "git@github.com:"
+        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+    - name: Download Ninja and CMake
+      id: cmake_and_ninja
+      shell: cmake -P {0}
```

Comment:
why is cmake used everywhere as the scripting language? doesn't that make it extra hard to make work?